src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py

```diff
@@ -225,6 +225,29 @@ def set_aligned_segment_starting_times(self, aligned_segment_starting_times: Lis
             ]
             self.set_aligned_segment_timestamps(aligned_segment_timestamps=aligned_segment_timestamps)
 
+    def set_probe(self, probe, *, group_mode: Literal["by_shank", "by_probe"]):
```

Comment:
```suggestion
    def set_probe(self, probe, group_mode: Literal["by_shank", "by_probe"]):
```

We don't tend to hard enforce keyword argument usage with input validation; if we started to, we should do it elsewhere but it would need to be discussed with the rest of the team

Reply:
I think enforcing this is a good idea. Let's discuss this on the next team meeting indeed.

Reply:
I think it is dangerous to not enforce it, for future-proofing backward compatibility. But for the sake of this PR I have removed the `*`.